src/views/Game/GameHooks.ts

```diff
@@ -49,52 +88,25 @@ export function useShowUndoRequested(goban: GobanCore): boolean {
 }
 
 /** React hook that returns true if user is a participant in this game */
-export function useUserIsParticipant(goban?: GobanCore) {
-    const [user_is_participant, setUserIsParticipant] = React.useState(false);
-    React.useEffect(() => {
-        if (!goban) {
-            return;
-        }
-
-        setUserIsParticipant(goban.engine.isParticipant(data.get("user").id));
-        goban.on("load", () =>
-            setUserIsParticipant(goban.engine.isParticipant(data.get("user").id)),
-        );
-    }, [goban]);
-    return user_is_participant;
-}
+export const useUserIsParticipant = generateGobanHook((goban: GobanCore | null) => {
+    const user = data.get("user");
+    if (!goban || !user) {
+        return false;
+    }
+    return goban.engine.isParticipant(user.id);
+});
 
 /** React hook that returns the current move number from goban */
-export function useCurrentMoveNumber(goban: GobanCore): number {
-    const [cur_move_number, setCurMoveNumber] = React.useState(
-        goban.engine.cur_move?.move_number || -1,
-    );
-    React.useEffect(() => {
-        goban.on("load", () => setCurMoveNumber(goban.engine.cur_move?.move_number || -1));
-        goban.on("cur_move", (move) => setCurMoveNumber(move.move_number));
-    }, [goban]);
-    return cur_move_number;
-}
+export const useCurrentMoveNumber = generateGobanHook(
+    (goban: GobanCore) => goban.engine.cur_move?.move_number || -1,
```

Comment:
I figured with this being a `(goban: GobanCore) => number` hook before, I should maintain this behavior in a refactor.  `generateGobanHook` can handle this because `GobanCore` extends `GobanCore | null`, and it returns a hook that only accepts a non-null Goban.

That said, maybe all Goban-related code should assume nullness? In most sub-components, I do assume `goban` is not null so I'm not sure.

Reply:
I think it is a safe assumption that goban will not be `null` in subcomponents, I'm pretty sure the Game page still early returns `null` if `goban` is null, and waits for it to be created before attempting to create subcomponents, and I see no reason to litter the subcomponents with unnecessary null checks.

In this case it just caught my eye that the other hooks were doing null checks and this one wasn't, didn't know if it needed it or not.